lib/ansible/modules/cloud/amazon/dms_replication_subnet_group.py

```diff
@@ -0,0 +1,246 @@
+        ReplicationSubnetGroupDescription=module.params.get('description'),
+        SubnetIds=module.params.get('subnetids'),
+    )
+
+    return instance_parameters
+
+
+def compare_params(param_described):
+    """
+    Compares the dict obtained from the describe function and
+    what we are reading from the values in the template We can
+    never compare passwords as boto3's method for describing
+    a DMS endpoint does not return the value for
+    the password for security reasons ( I assume )
+    """
+    modparams = create_module_params()
+    changed = False
+    # need to sanitize values that get retured from the API
+    if 'VpcId' in param_described.keys():
+        param_described.pop('VpcId')
+    if 'SubnetGroupStatus' in param_described.keys():
+        param_described.pop('SubnetGroupStatus')
+    for paramname in modparams.keys():
+        if paramname in param_described.keys() and \
+                param_described.get(paramname) == modparams[paramname]:
+            pass
+        elif paramname == 'SubnetIds':
+            subnets = []
+            for subnet in param_described.get('Subnets'):
+                subnets.append(subnet.get('SubnetIdentifier'))
+            for modulesubnet in modparams['SubnetIds']:
+                if modulesubnet in subnets:
+                    pass
+        else:
+            changed = True
+    return changed
+
+
+def create_replication_subnet_group(connection):
+    try:
+        params = create_module_params()
+        return replication_subnet_group_create(connection, **params)
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg="Failed to create DMS replication subnet group.",
+                         exception=traceback.format_exc(),
+                         **camel_dict_to_snake_dict(e.response))
+    except botocore.exceptions.BotoCoreError as e:
+        module.fail_json(msg="Failed to create DMS replication subnet group.",
+                         exception=traceback.format_exc())
+
+
+def modify_replication_subnet_group(connection):
+    try:
+        modify_params = create_module_params()
+        return replication_subnet_group_modify(connection, **modify_params)
+    except botocore.exceptions.ClientError as e:
+        module.fail_json(msg="Failed to Modify the DMS replication subnet group.",
+                         exception=traceback.format_exc(),
+                         **camel_dict_to_snake_dict(e.response))
+    except botocore.exceptions.BotoCoreError as e:
+        module.fail_json(msg="Failed to Modify the DMS replication subnet group.",
+                         exception=traceback.format_exc())
+
+
+def delete_replication_subnet_group(connection):
+    return True
+
+
+def main():
+    argument_spec = dict(
+        state=dict(choices=['present', 'absent'], default='present'),
+        identifier=dict(required=True),
+        description=dict(required=True),
+        subnetids=dict(type='list', required=True),
```

Comment:
```suggestion
        subnetids=dict(type='list', required=True, elements='str'),
```
Except if you want the list elements to be of another type.